### PR TITLE
Add more tests

### PR DESCRIPTION
**Pull request by rj-lee**
- Fix a bug where you get stuck infinitely at the end of a mine after a battle (thanks, `test_victory`)
- Change an armory menu option: if you choose a slot and have no equipment of that type, you will go back to selecting a slot for the currently selected character instead of going back out to the armory character selection menu
- Add more tests and organize them

### CODE REVIEW

**Comment by rj-lee** [comment]
The tests actually run the game; we can have even greater confidence that the game works as expected and they have helped to catch some bugs

Screenshot 2023-08-26 at 19 50 12

**Comment by rj-lee** [comment]
One downside of a full-on integration test like this is that the tests aren't very fast, there seems to be a delay between each key press. And of course the mining tests happen in real time

**Review by TheRedPanda17 — COMMENTED** [review]
Really great stuff. Just one issue to address

File: myning/chapters/armory.py [hunk 1]
```diff
@@ -80,11 +80,10 @@ def pick_equipment(c: Character, slot: ItemType):
 
 
 def equip(c: Character, equipment: Item):
-    if equipped := c.equipment.get_slot_item(equipment.type):
-        inventory.add_item(equipped)
     inventory.remove_item(equipment)
```
[the remark this answers is not in the document]
Comment (rj-lee) [reply myning/chapters/armory.py:83]: we take the equipment out of the inventory and put it on the character

File: myning/chapters/armory.py [hunk 2]
```diff
@@ -98,6 +97,6 @@ def auto_equip():
                 if equipped:
                     inventory.add_item(equipped)
                 inventory.remove_item(best)
-                character.equipment.change_item(best)
-    FileManager.multi_save(*player.army, inventory)
+                character.equipment.equip(best)
+    FileManager.multi_save(*player.army)
```
[the remark this answers is not in the document]
Comment (rj-lee) [reply myning/chapters/armory.py:101]: good catch, might be mistake in rebase from the inventory separation migration

File: myning/objects/equipment.py [hunk 3]
```diff
@@ -26,11 +26,13 @@ def all_items(self):
     def clear(self):
         self._slots = {category: None for category in EQUIPMENT_TYPES}
 
-    def get_slot_item(self, slot):
+    def get_slot_item(self, slot: ItemType):
         return self._slots[slot]
 
-    def equip(self, slot, item):
-        self._slots[slot] = item
+    def equip(self, item: Item):
```
[the remark this answers is not in the document]
Comment (rj-lee) [reply myning/objects/equipment.py:32]: sure

File: tests/chapters/test_armory.py [hunk 4]
```diff
@@ -0,0 +1,46 @@
...
+player = Player()
+inventory = Inventory()
+
+
+async def test_armory(pilot: Pilot, chapter: ChapterWidget):
+    player.upgrades.append(UPGRADES["armory_hints"])
+    player.upgrades.append(UPGRADES["auto_equip"])
+
+    # no weapon
+    await pilot.press("a")
```
[the remark this answers is not in the document]
Comment (rj-lee) [reply tests/chapters/test_armory.py:18]: run tests with `pytest --headed` to see it in action! Adding this to the readme

File: myning/chapters/armory.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (rj-lee) [reply myning/chapters/armory.py:83]: i see your other comment now. i'll revert the unequipping logic

**Review by TheRedPanda17 — APPROVED** [review]

**Pull request merged**